local-tasks.cake

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) The NUnit Project and distributed under the MIT License
+
+// This file contains tasks intended to be run locally by developers
+// in any NUnit project. To use these tasks in a project, copy the file
+// to an accessible directory and #load it in your build.cake file.
+
+// Since the tasks are intended for use by the NUnit Project, it follows
+// certain conventions and will need to be modified for use elsewhere.
+
+//////////////////////////////////////////////////////////////////////
+// DELETE ALL OBJ DIRECTORIES
+//////////////////////////////////////////////////////////////////////
+
+Task("DeleteObjectDirectories")
+    .WithCriteria(BuildSystem.IsLocalBuild)
+    .Does(() =>
+    {
+        Information("Deleting object directories");
+
+        foreach (var dir in GetDirectories("src/**/obj/"))
+            DeleteDirectory(dir, new DeleteDirectorySettings() { Recursive = true });
+    });
+
+// NOTE: Any project to which this file is added is required to have a 'Clean' target
+Task("CleanAll")
+    .Description("Perform standard 'Clean' followed by deleting object directories")
+    .IsDependentOn("Clean")
+    .IsDependentOn("DeleteObjectDirectories");
+
+//////////////////////////////////////////////////////////////////////
+// CHECK FOR MISSING AND NON-STANDARD FILE HEADERS
+//////////////////////////////////////////////////////////////////////
+
+static readonly int CD_LENGTH = Environment.CurrentDirectory.Length + 1;
+
+// Standard Header. Change this for each project as needed.
+static readonly string[] STD_HDR = new [] {
+    "// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt"
+};
+
+Task("CheckHeaders")
+    .WithCriteria(BuildSystem.IsLocalBuild)
```

Comment:
I wonder about adding this to CI as a check, or maybe also to the TestAll target? My thinking is it would be good if new contributors hit this automatically before they did a pull request - so it's gets sorted from the start rather than us having to ask people to amend. We'd need to make the task throw an error, of course.

Reply:
I can do that but I feel like we need a stronger capability to eliminate files we do __not__ want checked, e.g. Options.cs. I'll add a pattern of files to __not__ check in addition to the ones we check and eliminate the criterion.